docs/proposals/01-extensibility.md

```diff
@@ -0,0 +1,879 @@
+| botanist.DeployExternalDomainDNSRecord | Gardener creates a DNS-specific CRD in the Seed, and the responsible DNS-controller picks it up and creates a corresponding DNS record: (see CRD specification above). |
+| shootCloudBotanist.DeployInfrastructure | Gardener creates a Infrastructure-specific CRD in the Seed, and the responsible Botanist picks it up and does its job: (see CRD above). |
+| botanist.DeployBackupInfrastructure | Gardener creates a `BackupInfrastructure` resource in the Garden cluster.<br>(The BackupInfrastructure controller creates a BackupInfrastructure-specific CRD in the Seed, and the responsible Botanist picks it up and does its job: (see CRD above).) |
+| botanist.WaitUntilBackupInfrastructureReconciled | Gardener checks the `.status` object of the just created `BackupInfrastructure` resource. |
+| hybridBotanist.DeployETCD | Gardener does only deploy the etcd `StatefulSet` without backup-restore sidecar at all.<br>The cloud-specific Botanist registers a Mutating Webhook and adds the backup-restore sidecar, and it also creates the `Secret` needed by the backup-restore sidecar. |
+| botanist.WaitUntilEtcdReady | Gardener checks the `.status` object of the etcd `Statefulset` and waits until readiness is indicated. |
+| hybridBotanist.DeployCloudProviderConfig | Gardener does not execute this anymore because it doesn't know anything about cloud-specific configuration. |
+| hybridBotanist.DeployKubeAPIServer | Gardener does only deploy the kube-apiserver `Deployment` without any cloud-specific flags/configuration.<br> The cloud-specific Botanist registers a Mutating Webhook and adds whatever is needed for the kube-apiserver to run in its cloud environment. |
+| hybridBotanist.DeployKubeControllerManager | Gardener does only deploy the kube-controller-manager `Deployment` without any cloud-specific flags/configuration.<br>The cloud-specific Botanist registers a Mutating Webhook and adds whatever is needed for the kube-controller-manager to run in its cloud environment (e.g., the cloud-config). |
+| hybridBotanist.DeployKubeScheduler | Gardener does only deploy the kube-scheduler `Deployment` without any cloud-specific flags/configuration.<br>The cloud-specific Botanist registers a Mutating Webhook and adds whatever is needed for the kube-scheduler to run in its cloud environment. |
+| hybridBotanist.DeployCloudControllerManager | Gardener does not execute this anymore because it doesn't know anything about cloud-specific configuration. The Botanists would be responsible to deploy their own cloud-controller-manager now.<br>They would watch for the kube-apiserver Deployment to exist, and as soon as it does, they deploy the CCM.<br> (Side note: The Botanist would also be responsible to deploy further controllers needed for this cloud environment, e.g. F5-controllers or CSI plugins). |
```

Comment:
Good question, thank you. Actually, with the described approach there is no way for reporting that. One way to solve that and to allow the external controllers to report such messages would be to introduce a dedicated `ControlPlane` resource. It wouldn't have any `spec` fields (because we will start with letting Gardener be responsible for the control plane), but it might also help us later in case we want to move this responsibility from Gardener to other components:

```yaml
---
apiVersion: extensions.gardener.cloud/v1alpha1
kind: ControlPlane
metadata:
  name: controlplane
  namespace: shoot--core--aws-01
spec: {}
status:
  observedGeneration: ...
  state: ...
  conditions: ....
  lastError: ..
  lastOperation: ...
```

Gardener would write it and wait until the `lastOperation` indicates success, and it also allows to report conditions/health checks for the components not known by Gardener.

WDYT?